runtime/near-wallet-contract/implementation/wallet-contract/src/internal.rs

```diff
@@ -216,15 +319,7 @@ fn validate_tx_relayer_data(
         return Err(Error::Relayer(RelayerError::InvalidNonce));
     }
 
-    // If `to == target` and this is not a self-transaction then the address must not
-    // be registered in the address registry. The purpose of this check is to prevent
-    // lazy relayers from skipping this check themselves (relayers are supposed to use
-    // the address registry to fill in the `target`).
```

Comment:
> If `to == target` and this is not a self-transaction then the address must not be registered in the address registry.

Why is that the case? Asking even though the comment is removed since I'm having some difficulties in understanding the changes in this function.

> the purpose of this check is to prevent lazy relayers from skipping this check

Since this check is removed, what prevents relayers from getting away with being lazy?

Reply:
Thank you for the explanation. Looking at the code only I think it's hard to connect the dots. It might help to mention that the algorithm described in the [comment](https://github.com/aurora-is-near/nearcore/blob/eth-wallet-nep-141-deposit/runtime/near-wallet-contract/implementation/wallet-contract/src/internal.rs#L48) is implemented not only in the code that follows, but also in `validate_tx_relayer_data` which is called above.

Reply:
A note about this is added in https://github.com/near/nearcore/pull/11435/commits/e5142cc526e870fffac11a2538667fa8a3565f9f